key.go

```diff
@@ -0,0 +1,791 @@
+	// private key fields.
+	KeyOpDeriveBits KeyOp = 8
+
+	// The key is used for creating MACs.
+	KeyOpMACCreate KeyOp = 9
+
+	// The key is used for validating MACs.
+	KeyOpMACVerify KeyOp = 10
+)
+
+// KeyOp represents a key_ops value used to restrict purposes for which a Key
+// may be used.
+type KeyOp int64
+
+// KeyOpFromString returns the KeyOp corresponding to the specified name.
+// The values are taken from https://www.rfc-editor.org/rfc/rfc7517#section-4.3
+func KeyOpFromString(val string) (KeyOp, error) {
+	switch val {
+	case "sign":
+		return KeyOpSign, nil
+	case "verify":
+		return KeyOpVerify, nil
+	case "encrypt":
+		return KeyOpEncrypt, nil
+	case "decrypt":
+		return KeyOpDecrypt, nil
+	case "wrapKey":
+		return KeyOpWrapKey, nil
+	case "unwrapKey":
+		return KeyOpUnwrapKey, nil
+	case "deriveKey":
+		return KeyOpDeriveKey, nil
+	case "deriveBits":
+		return KeyOpDeriveBits, nil
+	default:
+		return KeyOpInvalid, fmt.Errorf("unknown key_ops value %q", val)
+	}
+}
+
+func (ko KeyOp) String() string {
+	switch ko {
+	case KeyOpSign:
+		return "sign"
+	case KeyOpVerify:
+		return "verify"
+	case KeyOpEncrypt:
+		return "encrypt"
+	case KeyOpDecrypt:
+		return "decrypt"
+	case KeyOpWrapKey:
+		return "wrapKey"
+	case KeyOpUnwrapKey:
+		return "unwrapKey"
+	case KeyOpDeriveKey:
+		return "deriveKey"
+	case KeyOpDeriveBits:
+		return "deriveBits"
+	case KeyOpMACCreate:
+		return "MAC create"
+	case KeyOpMACVerify:
+		return "MAC verify"
+	default:
+		return "unknown key_op value " + strconv.Itoa(int(ko))
+	}
+}
+
+// IsSupported returnns true if the specified value is represents one of the
+// key_ops defined in
+// https://www.rfc-editor.org/rfc/rfc9052.html#name-cose-key-common-parameters
+func (ko KeyOp) IsSupported() bool {
+	return ko >= 1 && ko <= 10
+}
+
+// MarshalCBOR marshals the KeyOp as a CBOR int.
+func (ko KeyOp) MarshalCBOR() ([]byte, error) {
+	return encMode.Marshal(int64(ko))
+}
+
+// UnmarshalCBOR populates the KeyOp from the provided CBOR value (must be int
+// or tstr).
+func (ko *KeyOp) UnmarshalCBOR(data []byte) error {
+	var raw intOrStr
+
+	if err := raw.UnmarshalCBOR(data); err != nil {
+		return fmt.Errorf("invalid key_ops value %w", err)
+	}
+
+	if raw.IsString() {
+		v, err := KeyOpFromString(raw.String())
+		if err != nil {
+			return err
+		}
+
+		*ko = v
+	} else {
+		v := raw.Int()
+		*ko = KeyOp(v)
+
+		if !ko.IsSupported() {
+			return fmt.Errorf("unknown key_ops value %d", v)
+		}
+	}
+
+	return nil
+}
+
+// KeyType identifies the family of keys represented by the associated Key.
+// This determines which files within the Key must be set in order for it to be
+// valid.
+type KeyType int64
+
+const (
+	// Invlaid key type
+	KeyTypeInvalid KeyType = 0
+	// Octet Key Pair
+	KeyTypeOKP KeyType = 1
+	// Elliptic Curve Keys w/ x- and y-coordinate pair
+	KeyTypeEC2 KeyType = 2
+	// Symmetric Keys
+	KeyTypeSymmetric KeyType = 4
+)
+
+func (kt KeyType) String() string {
```

Comment:
Can you document this method noticing that it is for human reading only to prevent any misuse?